### PR TITLE
Change: Ability to convert between road types for town roads and bridges now indi…

### PR DESCRIPTION
**Pull request by UnsuspiciousGooball**
…rectly depends on town ratings.

## Motivation / Problem

Previously it was possible to convert town roads and bridges without restriction by the local authority.

## Description

With this change the ability to convert town roads and bridges depends on the players' companies' standing with the local authority, which is determined using CheckforTownRating() with the ROAD_REMOVE and the TUNNELBRIDGE_REMOVE TownRatingCheckTypes, respectively.

Basically the player is allowed to convert town roads if and only if the player would be allowed to destroy town roads, and the same goes for town bridges.

### CODE REVIEW

**Comment by LordAro** [comment]
Not that we particularly mind, but the commit author does not appear to match your GitHub user. Might want to fix that

**Review by LordAro — CHANGES_REQUESTED** [review]
This appears to do exactly the same thing at the top of both if/else branches.
Instead of duplicating quite so much, I think a better solution would be to pull it out of the loop, and just use a ternary (`tt == MP_TUNNELBRIDGE ? TUNNELBRIDGE_REMOVE : ROAD_REMOVE`) for the only bit that actually differs. Would be able to get rid of the `is_owner_town` variable too.

**Review by LordAro — APPROVED** [review]
LGTM
